dali/test/python/test_dali_tf_dataset_mnist.py

```diff
@@ -0,0 +1,494 @@
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import nvidia.dali as dali
+import tensorflow as tf
+import nvidia.dali.plugin.tf as dali_tf
+import os
+import numpy as np
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+from test_utils_tensorflow import *
+from shutil import rmtree as remove_directory
+
+from nose.tools import with_setup
+
+try:
+    from tensorflow.compat.v1 import Session
+    from tensorflow.compat.v1 import placeholder
+    from tensorflow.compat.v1 import truncated_normal
+    from tensorflow.compat.v1 import reset_default_graph
+    from tensorflow.compat.v1 import variable_scope
+    from tensorflow.compat.v1 import layers
+    from tensorflow.compat.v1 import global_variables_initializer
+    from tensorflow.compat.v1.data import make_initializable_iterator
+except:
+    # Older TF versions don't have compat.v1 layer
+    from tensorflow import Session
+    from tensorflow import placeholder
+    from tensorflow import truncated_normal
+    from tensorflow import reset_default_graph
+    from tensorflow import variable_scope
+    from tensorflow import layers
+    from tensorflow import global_variables_initializer
+    from tensorflow.data import make_initializable_iterator
+try:
+    from tensorflow.train import AdamOptimizer as Adam
+    from tensorflow.train import AdamOptimizer as AdamOptimizer
+except:
+    from tensorflow.compat.v1.keras.optimizers import Adam
+    from tensorflow.compat.v1.train import AdamOptimizer
+
+try:
+    tf.compat.v1.disable_eager_execution()
+except:
+    pass
+
+target = 0.85
```

Comment:
Maybe we can use capital letters for this global variables to make them distinct?

Reply:
Done